src/main/java/com/epam/task/third/logic/TriangleCharacteristicsCalculator.java

```diff
@@ -0,0 +1,131 @@
+package com.epam.task.third.logic;
+
+import com.epam.task.third.entities.Point;
+import com.epam.task.third.entities.Triangle;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TriangleCharacteristicsCalculator {
+
+    public TriangleCharacteristicsCalculator() {
+    }
+
+    private List<Double> calculateSides(List<Point> points){
+        Point firstPoint = points.get(0);
+        Point secondPoint = points.get(1);
+        Point thirdPoint = points.get(2);
+        List<Double> sides = new ArrayList<>();
+        Double firstSide = Math.sqrt(Math.pow(firstPoint.getX() - secondPoint.getX(), 2) +
+                Math.pow(firstPoint.getY() - secondPoint.getY(), 2));
+        sides.add(firstSide);
+        Double secondSide = Math.sqrt(Math.pow(firstPoint.getX() - thirdPoint.getX(), 2) +
+                Math.pow(firstPoint.getY() - thirdPoint.getY(), 2));
+        sides.add(secondSide);
+        Double thirdSide = Math.sqrt(Math.pow(secondPoint.getX() - secondPoint.getX(), 2) +
+                Math.pow(secondPoint.getY() - secondPoint.getY(), 2));
+        sides.add(thirdSide);
+        return sides;
+    }
+
+    public double findPerimeter(Triangle triangle) {
+        List<Point> points = triangle.getCoordinates();
+        List<Double> sides = calculateSides(points);
+        return sides.get(0) + sides.get(1) + sides.get(2);
+    }
+
+    public double findArea(Triangle triangle) {
+        List<Point> points = triangle.getCoordinates();
+        List<Double> sides = calculateSides(points);
+        Double semiPerimeter = (sides.get(0) + sides.get(1) + sides.get(2)) / 2;
+        return Math.sqrt(semiPerimeter * (semiPerimeter - sides.get(0)) *
+                (semiPerimeter - sides.get(1)) *
+                (semiPerimeter - sides.get(2)));
+    }
+
+    public boolean isTriangleValid(Triangle triangle) {
```

Comment:
validate Triangle points before triangle creation in a different validator class